src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -115,6 +115,11 @@ private Task<Socket> AcceptAsyncApm(Socket acceptSocket)
 
         internal Task ConnectAsync(EndPoint remoteEP)
         {
+            // Throw up front if we are already connected
+            if (this.Connected)
+            {
```

Comment:
Since this and the other methods in this file are just wrappers to BeginConnect, I don't think we need to check this here, do we?

Reply:
Good call. Do you think we need to add the check to the other versions of BeginConnect, such as https://github.com/dotnet/corefx/blob/40cbd5636851093309b841db4d53af369b7ef1c6/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L2131? I only added it to the code paths we hit in the test cases so far.

Reply:
Yes, we should be consistent about doing the check in all variations of Connect/BeginConnect.